metloom/pointdata/geosphere_austria.py

```diff
@@ -0,0 +1,395 @@
+    URL = "https://dataset.api.hub.geosphere.at"
+    DATASOURCE = "GEOSPHERE"
+    META_EXTENSION = "/v1/station/current/tawes-v1-10min/metadata"
+
+    def _data_request(self, params):
+        """
+        Make get request and return JSON
+        Args:
+            params: dictionary of request parameters
+        Returns:
+            dictionary of response values
+        """
+        url = self.URL + "/v1/station/historical/tawes-v1-10min"
+        resp = requests.get(url, params=params)
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_daily_data(
+        self,
+        start_date: datetime,
+        end_date: datetime,
+        variables: List[SensorDescription],
+    ):
+        """
+        See docstring for PointData.get_daily_data
+        Example query:
+        https://dataset.api.hub.geosphere.at/v1/station/current/
+        tawes-v1-10min?parameters=TL&station_ids=11035
+        """
+        return self._get_data(start_date, end_date, variables, "D")
+
+    def get_hourly_data(
+        self,
+        start_date: datetime,
+        end_date: datetime,
+        variables: List[SensorDescription],
+    ):
+        """
+        See docstring for PointData.get_hourly_data
+        """
+        return self._get_data(start_date, end_date, variables, "H")
+
+
+class GeoSphereHistPointData(GeoSpherePointDataBase):
+    """
+    Implement PointData methods for GeoSphere Austria data source
+    API documentation here
+    https://dataset.api.hub.geosphere.at/v1/docs/index.html
+    https://dataset.api.hub.geosphere.at/v1/docs/user-guide/resource.html
+
+    Datasets available here https://data.hub.geosphere.at/dataset/
+
+    We use klima-v1-1d data which consists of historical daily data.
+    There is historical hourly data, but the parameter names are different
+    and as such this has not bee implemented
+    """
+
+    ALLOWED_VARIABLES = GeoSphereHistVariables
+    URL = "https://dataset.api.hub.geosphere.at"
+    DATASOURCE = "GEOSPHERE"
+    META_EXTENSION = "/v1/station/historical/klima-v1-1d/metadata"
+
+    def _data_request(self, params):
+        """
+        Make get request and return JSON
+        Args:
+            params: dictionary of request parameters
+        Returns:
+            dictionary of response values
+        """
+        url = self.URL + "/v1/station/historical/klima-v1-1d"
+        resp = requests.get(url, params=params)
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_daily_data(
+        self,
+        start_date: datetime,
+        end_date: datetime,
+        variables: List[SensorDescription],
+    ):
+        """
+        See docstring for PointData.get_daily_data
+        Example query:
+        https://dataset.api.hub.geosphere.at/v1/station/historical/klima-v1-1d
+        ?station_ids=11401&start=2023-04-12&end=2023-04-14&parameters=schnee
+        """
+        return self._get_data(start_date, end_date, variables, None)
```

Comment:
Same question here as the current data. Should we catch requests that go past the time period?

Reply:
klima goes back as far as we want, it's just tawes that is the issue